api/v1alpha1/common_types.go

```diff
@@ -23,6 +23,14 @@ type ObjectReference struct {
 	Name string `json:"name"`
 }
 
+type TypedObjectReference struct {
+	// Kind is the type of resource being referenced
+	// +kubebuilder:validation:Required
+	Kind string `json:"kind"`
```

Comment:
@zachaller what are your thoughts about this vs. `clusterScmProviderRef`? This matches ESO, which is nice.

If we go this way, we should have an enum validation on this field.

Reply:
I like kind, it also matches what like RoleBindings due which I think we could also use that concept for here as well. I agree though with the string we should enable enum validation via cel.

Reply:
I think we can also default to ScmProvider vs ""

Reply:
I wanted to add an enum validation but this is in a generic `TypedObjectReference` so we can't really add it directly to the field, unless I make this struct specific to the ScmProviders

Reply:
I'd just make it specific, I think it's worth it for the validation.